File: dan/java/src/main/java/com/ayulit/task01/Sign.java
package com.ayulit.task01;

public class Sign implements IPrintable {

    private char sign;
    
    public Sign(char sign) {
        this.sign = sign;
    }
    
    // we can print 'sign' on some sort of printer
    @Override
    public void print(IPrinter printer) {
        printer.printString(Character.toString(sign));
    }

}

File: dan/cpp/DanDesignPatterns/sign.cpp
#include "sign.h"

Sign::Sign(char sign)
{
     qDebug() << "Sign object created" << endl;

     this->sign = sign;

     qDebug() << "this->sign=" << this->sign << endl;
}

// we can print 'sign' on some sort of printer
void Sign::print(IPrinter printer)
{
    // stub for a while
}
